mmv1/products/integrationconnectors/Connection.yaml

```diff
@@ -0,0 +1,1075 @@
+            description: |
+              The user account used to authenticate.
+          - !ruby/object:Api::Type::NestedObject
+            name: "sshClientCert"
+            description: |
+              Secret version reference containing the client secret.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: "secretVersion"
+                description: |
+                  The resource name of the secret version in the format,
+                  format as: projects/*/secrets/*/versions/*.
+          - !ruby/object:Api::Type::String
+            name: "certType"
+            description: |
+              Format of SSH Client cert.
+          - !ruby/object:Api::Type::NestedObject
+            name: "sshClientCertPass"
+            description: |
+              Secret version reference containing the client secret.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: "secretVersion"
+                description: |
+                  The resource name of the secret version in the format,
+                  format as: projects/*/secrets/*/versions/*.
+      - !ruby/object:Api::Type::NestedObject
+        name: "oauth2AuthCodeFlow"
+        properties:
+          - !ruby/object:Api::Type::String
+            name: "authCode"
+            description: |
+              Authorization code to be exchanged for access and refresh tokens.
+          - !ruby/object:Api::Type::String
+            name: "pkceVerifier"
+            description: |
+              PKCE verifier to be used during the auth code exchange.
+          - !ruby/object:Api::Type::String
+            name: "redirectUri"
+            description: |
+              Redirect URI to be provided during the auth code exchange.
+          - !ruby/object:Api::Type::String
+            name: "clientId"
+            description: |
+              Client ID for user-provided OAuth app.
+          - !ruby/object:Api::Type::NestedObject
+            name: "clientSecret"
+            description: |
+              Client secret for user-provided OAuth app.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: "secretVersion"
+                description: |
+                  The resource name of the secret version in the format,
+                  format as: projects/*/secrets/*/versions/*.
+          - !ruby/object:Api::Type::Array
+            name: "scopes"
+            description: |
+              Scopes the connection will request when the user performs the auth code flow.
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Boolean
+            name: "enablePkce"
+            description: |
+              Whether to enable PKCE when the user performs the auth code flow.
+          - !ruby/object:Api::Type::String
+            name: "authUri"
+            description: |
+              Auth URL for Authorization Code Flow.
```

Comment:
I believe these are for the union field `type` and at most one of them can be set. Therefore we may need `conflicts` flag. I've also realized there were multiple union fields above, but since there's bug in Terraform SDK that it won't work with lists, we can't add `conflicts` to them at this moment.